proposals/v3/reads.md

```diff
@@ -0,0 +1,183 @@
+
+Upstream of Synse Server, the context can be used as the consumer sees fit. A good use
+case for it would be to convert the k:v pairs in the context to Prometheus labels for
+that particular reading.
+
+The the section below for how the context field is represented in the response scheme.
+
+## Synse Server Read Response
+> See: [API: Read](api.md#read)
+
+The read response will need to change, both to support returning multiple readings as
+well as to provide additional reading context, if any exists. Directly below is an example
+of what a read response looks like in API v2, for reference.
+
+**v2 read response scheme example**
+```json
+{
+  "kind": "temperature",
+  "data": [
+    {
+      "value": 20.3,
+      "timestamp": "2018-02-01T13:47:40.395939895Z",
+      "unit": {
+        "symbol": "C",
+        "name": "degrees celsius"
+      },
+      "type": "temperature",
+      "info": ""
+    }
+  ]
+}
+```
+
+For the v3 read response scheme, we need to:
+* support multiple device read responses
+  * in addition, we need to include identifying information about the device to know which device
+    the reading is associated with. (previously this did not exist because synse-server new the
+    rack/board/device routing for a single device before it read from it, so it had that information
+    available -- now we will not, so it needs to be returned in the response).
+* optional context for a reading
+
+**v3 read response scheme example**
+```json
+{
+  "a72cs6519ee675b": [
+    {
+      "kind": "temperature",
+      "type": "temperature",
+      "value": 20.3,
+      "timestamp": "2018-02-01T13:47:40.395939895Z",
+      "info": "",
+      "unit": {
+        "symbol": "C",
+        "name": "degrees celsius"
+      },
+      "context": {
+        "host": "127.0.0.1",
+        "sample_rate": 8
+      }
+    }
+  ],
+  "929b923de65a811": [
+    {
+      "kind": "led",
+      "type": "state",
+      "value": "off",
+      "timestamp": "2018-02-01T13:47:40.395939895Z",
+      "info": "",
+      "unit": null
+    },
+    {
+      "kind": "led",
+      "type": "color",
+      "value": "000000",
+      "timestamp": "2018-02-01T13:47:40.395939895Z",
+      "info": "",
+      "unit": null
+    }
+  ],
+  "12bb12c1f86a86e": [
+    {
+      "kind": "door_lock",
+      "type": "lock",
+      "value": "locked",
+      "timestamp": "2018-02-01T13:47:40.395939895Z",
+      "info": "Zone 6B",
+      "unit": null,
+      "context": {
+        "wedge": 1
+      }
+    }
+  ]
+}
+```
+
+
+> **For Consideration**: 
+>
+> We could also simplify this (it is a lot of data to return, especially if we are
+> reading in bulk) by having some of the info be a separate lookup, e.g.
+>
+> ```
+> GET /read?tags="id=something"
+> {
+>   "a72cs6519ee675b": [
+>     "value": 20.3,
+>     "timestamp": "...",
+>     "type_id": 123235235134235,
+>     "context": {
+>       ...
+>     }
+>   ] 
+> }
+> ```
+>
+> ```
+> GET /type?id=123235235134235
```

Comment:
Does it mean that we need to have a new route `/type` to in order to query this? If that's the case then I think it's fine to leave thing as it is since it saves a little but then introduces another layer of complexity.

Reply:
this has been removed